### PR TITLE
[torch] Use core_aten_decompositions() for FX path

### PR DESCRIPTION
**Pull request by sahas3**
Switch `get_decomposition_table()` to use `core_aten_decompositions()` with a small exclusion list, rather than the hand-curated ~50 op table. This decomposes ~900 ops PT-side before FX import, making 62 C++ patterns in `DecomposeComplexOps` redundant on the FX path (these ops don't appear in IR for ONNX import path either).

The exclusion list (34 ops) covers `FFT` ops (no ODS def), `norm` ops (complex tensor regressions), and ops whose PT decomposition produces backend-incompatible IR (e.g., `roll` -> `arange+fmod+index_select`).

Results across all 4 e2e configs: 0 unexpected failures, 160 XPASS (newly passing tests from decompositions that had no C++ equivalent).

Implements https://github.com/llvm/torch-mlir/issues/4729

Assisted By: Claude Code.

### CODE REVIEW

File: python/torch_mlir/extras/fx_decomp_util.py [hunk 1]
```diff
@@ -57,4 +32,70 @@
...
+    torch.ops.aten.all,
+    torch.ops.aten.isfinite,
+    torch.ops.aten.logaddexp,
+    torch.ops.aten.logaddexp2,
+    torch.ops.aten.eye,
+    torch.ops.aten.isclose,
+]
+
+
+def get_expanded_decomposition_table():
```
Comment (penguin-wwy) [inline python/torch_mlir/extras/fx_decomp_util.py:85]: Maybe adding `@functools.cache` annotation

Comment (sahas3) [reply python/torch_mlir/extras/fx_decomp_util.py:85]: Done.

**Review by ziereis — CHANGES_REQUESTED** [review]
What does this mean for tm_tensor ops that will get converted to other high level ops? I think with this change here we would start decomposing the `aten.scaled_dot_product_attention` op which is something we can lower to tm_tensor.attention -> iree_linalg_ext.attention. 

Can we also add ops that would have an affect on the tm_tensor lowerings to the excluded list?

**Comment by sahas3** [comment]
> think with this change here we would start decomposing the `aten.scaled_dot_product_attention` op which is something we can lower to tm_tensor.attention -> iree_linalg_ext.attention.

Good catch -- you're right. Switching to `core_aten_decompositions()` does decompose `aten.scaled_dot_product_attention` at FX import time, which is a regression relative to `main`.

However, `sdpa` not decomposing in `main` is *maybe* unintentional since we already have in `main` https://github.com/llvm/torch-mlir/blob/0029fc2e102c6d604696b33eb618116b4957a2e3/python/torch_mlir/extras/fx_decomp_util.py#L55

The reason the op is not decomposed even though it's added to the list is because `get_decompositions()` builds its result purely from the `post_autograd` decomposition registry: for an overload packet it only emits decompositions if some overload of that packet is registered there, otherwise it silently ignores. `scaled_dot_product_attention` is a composite-implicit-autograd op -- it has no overload in the post-autograd registry. So `get_decompositions([torch.ops.aten.scaled_dot_product_attention])` returns `{}` and the append is effectively dead: sdpa never lands in `main`'s table, survives FX import intact, and the C++ `torch-decompose-complex-ops` pass decomposes it only for backends that mark it illegal. That's how `TorchToTMTensor` keeps it as` tm_tensor.attention`. FYI @catcor01 since you added `sdpa` to `main`'s decomp list, I presume the motivation was to decompose it but that never actually happened. 

`core_aten_decompositions()` is different -- it does include composite-implicit-autograd ops, decomposing them by unfolding the composite implementation, so on this branch sdpa lands in the table and gets decomposed at import time, removing that per-backend choice.

I can add `scaled_dot_product_attention` to the exclude list, restoring `main`'s behavior. The TOSA/stablehlo masked/causal/GQA sdpa tests will then go back to xfail (same as `main` -- they never lowered there; they only started XPASSing on this branch because of the import-time decomposition).

@ziereis, what is your thought on leaving the `sdpa` decomp as it is now but as a consumer you can remove it from the decomp list as per your needs to preserve the op through decomposition?

```
import torch
from torch_mlir import fx
from torch_mlir.extras.fx_decomp_util import get_decomposition_table

table = dict(get_decomposition_table())          # copy — source is @cache'd
table.pop(torch.ops.aten.scaled_dot_product_attention.default, None)  # safe remove
m = fx.export_and_import(model, *inputs, decomposition_table=table)
```

> Can we also add ops that would have an affect on the tm_tensor lowerings to the excluded list?

I also audited the other 10 ops with dedicated `TorchToTMTensor` lowerings -- none are decomposed by `core_aten_decompositions()`, so sdpa was the only affected op.

**Review by ziereis — APPROVED** [review]
i think excluding it should work fine. Thanks for elaborating :)

**Review by penguin-wwy — APPROVED** [review]
lgtm
